nats.go

```diff
@@ -884,6 +884,32 @@ func RootCAs(file ...string) Option {
 	}
 }
 
+// RootCAsMem is a helper option to provide the RootCAs pool from memory.
+// If Secure is not already set this will set it as well.
+func RootCAsMem(rootCAs []byte) Option {
+	return func(o *Options) error {
+		rootCAsCB := func() (*x509.CertPool, error) {
+			pool := x509.NewCertPool()
+			for _, f := range rootCAs {
```

Comment:
As I mentioned before, this does not make sense, you are iterating over a byte slice, byte by byte, and growing cert pool for each byte...

Please look at this comment (which you marked as resolved): https://github.com/nats-io/nats.go/pull/1253#discussion_r1249472944

Reply:
ok, should I caste rootCAs to string here or convert the type to string?


func RootCAsMem(rootCAs string) Option {

or cast it here

for _, f := range string(rootCAs) {

or maybe it does not matter where, just looking for optimum solution.

It can be done at application level or here.

Reply:
Correct me if I am wrong, but it must be processed as a byte array to include line endings in the certificate. I tested, and it fails when cast to string, Perhaps there is a different method that I am un-aware of?  Pool does not have an append to add the line endings. Please advise on what to do,

Reply:
btw: byte array is way original code worked

Reply:
That is not what I meant, I do not think you should be casting it to string. You have one argument to function - `rootCAs []byte` - it is a single CA I assume. In `rootCAsCB` you are iterating **over a byte slice**, which means the loop will be executed `len(rootCAs)` times. As a simple example, try putting any `Println` in this for loop, e.g. like this:
```
rootCAsCB := func() (*x509.CertPool, error) {
	pool := x509.NewCertPool()
	for _, f := range rootCAs {
		fmt.Println("Appending cert from PEM")
		ok := pool.AppendCertsFromPEM(rootCAs)
		if !ok {
			return nil, fmt.Errorf("nats: failed to parse root certificate from memory from %q", f)
		}
	}
	return pool, nil
}
``` 
And then execute `TestClientCertificateMem` and see what happens.

Reply:
Sorry to only get back to it now. Your `rootCAsCB` is still not correct I think and could be simplified to this:
```go
rootCAsCB := func() (*x509.CertPool, error) {
	pool := x509.NewCertPool()
	ok := pool.AppendCertsFromPEM(rootCAs)
	if !ok {
		return nil, fmt.Errorf("nats: failed to parse root certificate from memory")
	}
	return pool, nil
}
```

There is no need for the for loop. Also, the error message is not correct as `f` is a single byte.